### ISSUE TITLE
Request the MedSAM 2 weights

### ISSUE THREAD

**Issue by LiGuo12** [MedicineToken/Medical-SAM2#2]
Hi, could you release the trained weights of MedSAM 2? I noticed that you used SAM2-small. Have you tried the base and large versions? Can you release their weights?

**Comment by jiayuanz3** [comment]
We've tried both -s and -t, but it's also convenient to train on base and large version based on SAM2's configuration file (https://github.com/facebookresearch/segment-anything-2). 

Yes, we are preparing the pre-trained weight with demo and it will be released on hugging face soon. Do you specifically mean 2D or 3D weight?

**Comment by LiGuo12** [comment]
Thank you for your reply. I am interested in the 2d -s version.

Best,
Li

**Comment by LiGuo12** [comment]
Do you plan to train the SAM2-base version? Given MedSAM uses SAM-base, it may be better if you can train a MedSAM2 based on SAM2-base as a comparison.

Additionally, there is a potential issue with the scheduler implementation in the training code (train_2d.py and train_3d.py). It appears the scheduler may not be used correctly. Could you please review this?

**Comment by LiGuo12** [comment]
Hi, 

have you verified the code (train_2d.py)?
Based on your code, I fine-tuned on a dataset and the best dice of the saved weights was 61 (checkpoint['best_dice']), but when I tested with the saved weights, the results were very poor. Have you verified that the saving and loading weights process in train_2d.py is correct?

**Comment by li-pengcheng** [comment]
INFO:root:Total score: 0.09795794636011124, IOU: 0.7826385322415063, DICE: 0.8748504036664962 || @ epoch 36.
Total score: 0.09795794636011124, IOU: 0.7826385322415063, DICE: 0.8748504036664962 || @ epoch 36.
saving the best checkpoint at epoch 37 with dice: 0.8748504036664962

when i use the script in step2, I got best dice 0.87，but based on the paper, the best dice should be (97.8+87.6)/2 = 92.7, what am I missing?

**Comment by jiayuanz3** [comment]
These two results represent two distinct tasks, one for optic-disc and one for optic-cup.

**Comment by li-pengcheng** [comment]
> These two results represent two distinct tasks, one for optic-disc and one for optic-cup.

The script in step 2 is used to train optic-disc, right? My best result after running 100 epochs was 87.48.

**Comment by jiayuanz3** [comment]
It's for optic-cup

**Comment by jiayuanz3** [comment]
The saving and loading weights bug should be fixed now.
